src/ray/core_worker/transport/direct_task_transport.cc

```diff
@@ -51,10 +51,20 @@ Status CoreWorkerDirectTaskSubmitter::SubmitTask(TaskSpecification task_spec) {
               task_finisher_->CompletePendingTask(
                   task_id, push_task_reply, reply.actor_address());
             } else {
-              RAY_LOG(INFO) << "Failed to create actor " << actor_id
-                            << " with status: " << status.ToString();
+              rpc::RayErrorInfo ray_error_info;
+              if (status.IsSchedulingCancelled()) {
+                RAY_LOG(DEBUG) << "Actor creation cancelled, actor id = " << actor_id;
+                task_finisher_->MarkTaskCanceled(task_id);
+                ray_error_info.mutable_actor_died_error()->CopyFrom(reply.death_cause());
```

Comment:
if reply.has_death_cause()?

Reply:
It should always be set if the actor has been cancelled, should it be an `if` check or a `RAY_CHECK` in your opinion?

Reply:
I think it can be RAY_CHECK in this case?

Reply:
if might be a safer choice though